Anature/src/application/anatures/stats/Stats.java

```diff
@@ -253,35 +321,54 @@ private boolean levelGained()
-			return incrementLevel();
+			setLevel(nextLevel);
+			return levelStats();
 		}
+
 		return false;
 	}
 
-	private boolean incrementLevel()
+	private boolean levelStats()
 	{
-		mLevel = +1;
-
-		int hitPointsIncrese = (int) ((0.01 * getBaseHitPoints()) + (0.02 * (getIVHitPoints() + getEVHitPoints())));
-		addLevelHitPoints(hitPointsIncrese);
+		int hitPointsIncrease = (int) ((0.01 * (double) getBaseHitPoints()) + (0.02 * ((double) getIVHitPoints() + (double) getEVHitPoints())));
+		hitPointsIncrease = hitPointsIncrease == 0 ? 1 : hitPointsIncrease;
+		addLevelHitPoints(hitPointsIncrease);
+		maintainHitPointPercentage(hitPointsIncrease);
 
-		int attackIncrese = (int) ((0.01 * getBaseAttack()) + (0.02 * (getIVAttack() + getEVAttack())));
-		addLevelAttack(attackIncrese);
+		int attackIncrease = (int) ((0.01 * (double) getBaseAttack()) + (0.02 * ((double) getIVAttack() + (double) getEVAttack())));
+		attackIncrease = attackIncrease == 0 ? 1 : attackIncrease;
+		addLevelAttack(attackIncrease);
 
-		int defenseIncrese = (int) ((0.01 * getBaseDefense()) + (0.02 * (getIVDefense() + getEVDefense())));
-		addLevelDefnse(defenseIncrese);
+		int defenseIncrease = (int) ((0.01 * (double) getBaseDefense()) + (0.02 * ((double) getIVDefense() + (double) getEVDefense())));
+		defenseIncrease = defenseIncrease == 0 ? 1 : defenseIncrease;
+		addLevelDefnse(defenseIncrease);
 
-		int specialAttackIncrese = (int) ((0.01 * getBaseSpecialAttack()) + (0.02 * (getIVSpecialAttack() + getEVSpecialAttack())));
-		addLevelSpecialAttack(specialAttackIncrese);
+		int specialAttackIncrease = (int) ((0.01 * (double) getBaseSpecialAttack()) + (0.02 * ((double) getIVSpecialAttack() + (double) getEVSpecialAttack())));
+		specialAttackIncrease = specialAttackIncrease == 0 ? 1 : specialAttackIncrease;
+		addLevelSpecialAttack(specialAttackIncrease);
 
-		int specialDefenseIncrese = (int) ((0.01 * getBaseSpecialDefense()) + (0.02 * (getIVSpecialDefnese() + getEVSpecialDefense())));
-		addLevelSpecialDefense(specialDefenseIncrese);
+		int specialDefenseIncrease = (int) ((0.01 * (double) getBaseSpecialDefense())
+				+ (0.02 * ((double) getIVSpecialDefnese() + (double) getEVSpecialDefense())));
+		specialDefenseIncrease = specialDefenseIncrease == 0 ? 1 : specialDefenseIncrease;
+		addLevelSpecialDefense(specialDefenseIncrease);
 
-		int speedIncrese = (int) ((0.01 * getBaseSpeed()) + (0.02 * (getIVSpeed() + getEVSpeed())));
-		addLevelSpeed(speedIncrese);
+		int speedIncrease = (int) ((0.01 * (double) getBaseSpeed()) + (0.02 * ((double) getIVSpeed() + (double) getEVSpeed())));
+		speedIncrease = speedIncrease == 0 ? 1 : speedIncrease;
+		addLevelSpeed(speedIncrease);
 
 		return true;
 	}
+	
+	private void maintainHitPointPercentage(int hitPointsIncrease)
+	{
+		int currentHitPoitns = getCurrentHitPoints();
+		int previousTotalHitPoints = getTotalHitPoints() - hitPointsIncrease;
+		
+		double getHitPoitnsPercent = (double) currentHitPoitns / (double) previousTotalHitPoints;
```

Comment:
You wrote Poitns instead of Points for currentHitPoitns and getHitPoitnsPercent.